httpstan/services_stub.py

```diff
@@ -80,27 +81,41 @@ async def call(
         if arg not in kwargs:
             kwargs[arg] = typing.cast(typing.Any, arguments.lookup_default(arguments.Method[method.upper()], arg))
 
-    with socket.socket(socket.AF_UNIX, type=socket.SOCK_DGRAM) as socket_:
+    with socket.socket(socket.AF_UNIX, type=socket.SOCK_STREAM) as socket_:
         _, socket_filename = tempfile.mkstemp(prefix="httpstan_", suffix=".sock")
         os.unlink(socket_filename)
-        socket_.settimeout(0.001)
         socket_.bind(socket_filename)
+        socket_.listen(4)  # three stan callback writers, one stan callback logger
 
         lazy_function_wrapper = _make_lazy_function_wrapper(function_basename, model_name)
         lazy_function_wrapper_partial = functools.partial(lazy_function_wrapper, socket_filename.encode(), **kwargs)
         future = asyncio.get_running_loop().run_in_executor(executor, lazy_function_wrapper_partial)
 
+        potential_readers = [socket_]
         while True:
-            try:
-                message = socket_.recv(8192)
-            except socket.timeout:
+            # note: timeout of 0 required to avoid blocking
+            readable, writeable, errored = select.select(potential_readers, [], [], 0)
+            for s in readable:
+                if s is socket_:
+                    conn, _ = s.accept()
+                    potential_readers.append(conn)
+                    continue
+                message = s.recv(1024 * 256)
```

Comment:
Is this some specific lenght or just convention?

Reply:
If it's too short we risk splitting messages. 4096 is what the docs recommend but this will split multiple sequences of logger messages if they appear all at once (which they do with very fast sampling runs). 1024 * 256 is the default send buffer size on Linux so I picked that.